src/Scripting/Core/ScriptOptions.cs

```diff
@@ -92,6 +93,21 @@ private static ImmutableArray<MetadataReference> GetDefaultMetadataReferences()
         public ImmutableArray<string> Imports { get; private set; }
 
         /// <summary>
+        /// Specifies whether debugging symbols should be emitted.
+        /// </summary>
+        public bool EmitDebugInformation => DebugInformationFormat != null;
+
+        /// <summary>
+        /// <see cref="DebugInformationFormat"/> to be used for debugging symbols if needed
+        /// </summary>
+        public DebugInformationFormat? DebugInformationFormat { get; private set; } = null;
```

Comment:
That sounds like the best idea, I agree. I'm not sure what would be the determining factor though, is it similar to how it is the case for choosing the logic for assembly loading APIs? => if CoreCLR, use PortablePDB, otherwise use full ("properietary") PDB ?